src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
I just looked at the TDS spec for variant specification and it doesn't seem to match the code, at least as I read it. See '2.2.5.5.4 SQL_VARIANT Values' in the spec. Perhaps this code needs a bigger overhaul?